lib/ReactViews/FeatureInfo/FeatureInfoSection.jsx

```diff
@@ -371,4 +377,79 @@ function describeFromProperties(properties, time) {
     return html;
 }
 
+/**
+ * @param  {ReactClass} that The FeatureInfoSection.
+ * @return {Object} Returns {templateData, info, rawData, showRawData, hasRawData}.
+ *                  templateData is the object passed to the templating engine.
+ *                  info is the main body of the info section, as a react component.
+ *                  rawData is the same for the raw data, if it needs to be shown.
+ *                  showRawData is whether to show the rawData.
+ *                  hasRawData is whether there is any rawData to show.
+ */
+function getInfoAsReactComponent(that) {
+    const templateData = that.getPropertyValues();
+    const updateCounters = that.props.feature.updateCounters;
+    if (defined(templateData)) {
+        delete templateData._terria_columnAliases;
+    }
+    const context = {
+        catalogItem: that.props.catalogItem,
+        feature: that.props.feature,
+        updateCounters: updateCounters
+    };
+    const showRawData = !that.hasTemplate() || that.state.showRawData;
+    let rawDataHtml;
+    let rawData;
+    if (showRawData) {
+        rawDataHtml = that.descriptionFromFeature();
+        if (defined(rawDataHtml)) {
+            rawData = renderMarkdownInReact(rawDataHtml, context);
+        }
+    }
+    return {
+        templateData: templateData,
+        info: that.hasTemplate() ? renderMarkdownInReact(that.descriptionFromTemplate(), context) : rawData,
+        rawData: rawData,
+        showRawData: showRawData,
+        hasRawData: !!rawDataHtml
+    };
+}
+
+function setTimeoutsForUpdatingCustomComponents(that) { // eslint-disable-line require-jsdoc
+    const {info} = getInfoAsReactComponent(that);
+    const foundCustomComponents = CustomComponents.find(info);
+    foundCustomComponents.forEach((match, componentNumber) => {
+        const updateSeconds = match.type.selfUpdateSeconds(match.reactComponent);
+        if (updateSeconds > 0) {
+            setTimeoutForUpdatingCustomComponent(that, match.reactComponent, updateSeconds, componentNumber);
+        }
+    });
+}
+
+function setTimeoutForUpdatingCustomComponent(that, reactComponent, updateSeconds, componentNumber) { // eslint-disable-line require-jsdoc
```

Comment:
Ok, that makes sense.  It still bothers me a bit that this logic is here, though, especially given how complicated it is.  For example, if we wanted to add support for embedded charts on the Workbench, we'd have to duplicate all this logic, right?  I guess we can just plan to refactor later when we get to that, but it might be worth at least thinking about how that might work.